### PR TITLE
Remove de-duplication from Chisel

### PR DESCRIPTION
**Pull request by donggyukim**
@azidar @jackkoenig @chick @aswaterman @sdtwigg 
This PR is opened because module instance de-duplication will be implemented in FIRRTL.

### CODE REVIEW

File: chiselFrontend/src/main/scala/chisel3/core/Module.scala [hunk 1]
```diff
@@ -101,17 +101,11 @@ extends HasId {
   final val name = Builder.globalNamespace.name(desiredName)
 
   /** FIRRTL Module name */
-  private var _modName: Option[String] = None
-  private[chisel3] def setModName(name: String) = _modName = Some(name)
-  def modName = _modName match {
-    case Some(name) => name
-    case None => throwException("modName should be called after circuit elaboration")
-  }
+  def modName = name
```
Comment (aswaterman) [inline chiselFrontend/src/main/scala/chisel3/core/Module.scala]: Can we just delete modName?  It's not supposed to be part of the public API, anyway.

File: src/main/scala/chisel3/internal/firrtl/Emitter.scala [hunk 2]
```diff
@@ -83,17 +80,10 @@ private class Emitter(circuit: Circuit) {
...
-        m.id setModName duplicate.name
-        ""
-      case None =>
-        defnMap((m.id.desiredName, defn)) = m
-        m.id setModName m.name
-        moduleDecl(m) + defn
-    }
+    val sb = new StringBuilder
+    sb append moduleDecl(m)
+    sb append moduleDefn(m)
```
Comment (jackkoenig) [inline src/main/scala/chisel3/internal/firrtl/Emitter.scala]: These should use dot notation
Infix notation should only be used for purely functional methods (http://docs.scala-lang.org/style/method-invocation.html#arity-1)

**Comment by jackkoenig** [comment]
Even with approval, wait to merge this until the accompanying changes are merged to Firrtl so this can be retested and pass

**Comment by azidar** [comment]
I believe the changes have been made to Firrtl - can this be merged now?

**Comment by jackkoenig** [comment]
retest this please

**Comment by colinschmidt** [comment]
This breaks de-duplication in Hwacha. Chisel3/master firrtl/master works as expected but chisel3/remove-deduplication and firrtl/master fails to deduplicate properly. I'll investigate a little more and hopefully get closer to a root cause to help track down the issue.

**Comment by colinschmidt** [comment]
I found two Arbiter instances that are identical except for their name. Do we need to reopen the FIRRTL PR/issue ?

**Comment by colinschmidt** [comment]
See https://github.com/ucb-bar/firrtl/issues/371

**Comment by ucbjrl** [comment]
The breakage was caused by building with the onetrueliteral pr. Building
with master looks ok.

**Pull request merged**